indico/modules/rb_new/client/js/util.js

```diff
@@ -21,3 +21,32 @@ import moment from 'moment';
 export function toMoment(dt, format) {
     return dt ? moment(dt, format) : null;
 }
+
+export function parseRoomListFiltersText(text) {
+    const result = {text: null, building: null, floor: null};
+    if (!text) {
+        return result;
+    }
+
+    const reSplit = /((?:building:|floor:)(?:.*?)(?: |$))/;
+    const splitted = text.split(reSplit);
```

Comment:
"splitted" is *very* uncommon or even wrong/outdated in English (it's irregular - split/split/split).
I'd call it `parts` or something like that.